ocaml/xapi/xapi_vdi_helpers.ml

```diff
@@ -28,6 +28,18 @@ let assert_managed ~__context ~vdi =
   if not (Db.VDI.get_managed ~__context ~self:vdi)
   then raise (Api_errors.Server_error(Api_errors.vdi_not_managed, [ Ref.string_of vdi ]))
 
+(* CA-94283: Block VDI-Destroy operation when HA Enabling/Disabling is in progress *)
+let assert_ha_in_progress ~__context ~vdi =
+  let pool = Helpers.get_pool ~__context in
+  let ha_vdi = Db.VDI.get_name_label ~__context ~self:vdi in
+  if (ha_vdi = "Metadata for HA") || (ha_vdi = "Statefile for HA") then begin
```

Comment:
A better way to check for this is to look at the VDI type - it will be either `` `ha_statefile`` or `` `metadata``.